src/providers/experimentalWebViewProvider/ExperimentalWebviewProvider.ts

```diff
@@ -40,9 +39,8 @@ export class ExperimentalWebviewProvider implements IDappProvider {
   };
 
   logout = async () => {
-    const response = await this._provider.logout();
-    await logoutFromDApp();
-    return response;
+    store.dispatch(logoutAction());
```

Comment:
why not like before?

Reply:
because it would generate a loop
dApp calls explicitly logout function -> logout function calls provider.logout (this line here) -> provider logout function calls logout function who calls again the provider.logout...